lib/content-info/index.js

```diff
@@ -0,0 +1,384 @@
+
+        if (encodedFileName) {
+            fileName = decodeFileName(encodedFileName[Number_2], encodedFileName[Number_1]);
+        }
+
+        // If filename* is not present or unparseable, then we are checking for filename in header
+        if (!fileName) {
+            fileName = fileNameRegex.exec(dispositionHeader);
+            fileName && (fileName = fileName[Number_1]);
+
+            // check if file name is wrapped in double quotes
+            // file name can contain escaped characters if wrapped in quotes
+            if (fileName && fileName[Number_0] === doubleQuotes) {
+                // remove quotes and escapes
+                fileName = fileName
+                    .substr(Number_1, fileName.length - Number_2)
+                    .replace(quotedPairRegex, TOKEN_$1);
+            }
+        }
+
+        return fileName;
+    };
+
+
+module.exports = {
+
+    /**
+    * From response object, mime info and filename of the response content is taken
+    *
+    * @param {Response} response - response instance
+    *
+    * @returns {ResponseContentInfo} - Return contentInfo of the response
+    */
+    contentInfo: function (response) {
+        var contentType = response.headers.get(CONTENT_TYPE),
+            contentDisposition = response.headers.get(CONTENT_DISPOSITION),
+            mimeInfo = getMimeInfo(contentType, response.stream || response.body),
+            fileName = getFileNameFromDispositionHeader(contentDisposition),
+            extension = mimeInfo.extension,
+
+            /**
+             * @typedef Response~ResponseContentInfo
+             *
+             * @property {String} mimeType sanitized mime type
+             * @property {String} mimeFormat format for the identified mime type
+             * @property {String} charset the normalized character set
+             * @property {String} extension extension identified from the mime type
+             * @property {String} fileName file name extracted from disposition header
+             */
+            contentInfo = {};
+
+
+        // if file name is not present in the content disposition headers, use a default file name
+        if (!fileName) {
+            // add extension to default if present
+            extension ? (fileName = DEFAULT_RESPONSE_FILENAME + DOT + extension) :
+                (fileName = DEFAULT_RESPONSE_FILENAME);
+        }
+
+        // create a compacted list of content info from mime info and file name
+        mimeInfo.mimeType && (contentInfo.mimeType = mimeInfo.mimeType);
+        mimeInfo.mimeFormat && (contentInfo.mimeFormat = mimeInfo.mimeFormat);
+        mimeInfo.charset && (contentInfo.charset = mimeInfo.charset);
+        extension && (contentInfo.extension = extension);
+        fileName && (contentInfo.fileName = fileName);
+
+        return contentInfo;
+    },
+
+    // file name extraction regexes are extracted for vulnerability tests
+    regexes: {
```

Comment:
Why not use them from an object in declaration? That way, it will not be another thing to recall while adding regexes.